src/lingon_sylt.rs

```diff
@@ -397,102 +418,114 @@ particle_prop! { l_gfx_particle_end_alpha, end_alpha }
     l_time
+    "An absolute time messurement, but the start time is ill defined"
     [] -> Type::Float => {
         let time = game!().total_time() as f64;
-        Ok(Value::Float(time))
+        Ok(Float(time))
     },
 );
 
 
 sylt_macro::extern_function!(
     "sylt::lingon_sylt"
     l_random
+    "Returns a uniformly sampled random float between 0 and 1"
     [] -> Type::Float => {
-        Ok(Value::Float(Uniform.sample().into()))
+        Ok(Float(Uniform.sample().into()))
     },
 );
 
 sylt_macro::extern_function!(
     "sylt::lingon_sylt"
     l_random_range
-    [Value::Int(lo), Value::Int(hi)] -> Type::Int => {
-        Ok(Value::Int(*lo + (Uniform.sample() * ((hi - lo) as f32)) as i64))
+    "Returns a randomized integer in the given range"
+    [One(Int(lo)), One(Int(hi))] -> Type::Int => {
+        Ok(Int(*lo + (Uniform.sample() * ((hi - lo) as f32)) as i64))
+    },
+    [Two(Int(lo), Int(hi))] -> Type::Int => {
+        Ok(Int(*lo + (Uniform.sample() * ((hi - lo) as f32)) as i64))
     },
 );
 
 
 sylt_macro::extern_function!(
     "sylt::lingon_sylt"
     l_bind_key
-    [Value::String(key), Value::String(name)] -> Type::Void => {
+    "Binds a keyboard key to an input name"
+    [One(String(key)), One(String(name))] -> Type::Void => {
         let key = if let Some(key) = Keycode::from_name(key) {
             key
         } else {
             return error!("l_bind_key", "'{}' is an invalid key", key);
         };
 
         use lingon::input::{Device::Key, Keycode};
-        game!().input.bind(Key(key), String::clone(name));
+        game!().input.bind(Key(key), std::string::String::clone(name));
 
-        Ok(Value::Nil)
+        Ok(Nil)
     },
 );
 
 sylt_macro::extern_function!(
     "sylt::lingon_sylt"
     l_bind_quit
-    [Value::String(name)] -> Type::Void => {
+    "Binds the windows quit action (pressing the X in the corner) - plus points if you make it jump"
+    [One(String(name))] -> Type::Void => {
         use lingon::input::Device::Quit;
-        game!().input.bind(Quit, String::clone(name));
-        Ok(Value::Nil)
+        game!().input.bind(Quit, std::string::String::clone(name));
+        Ok(Nil)
     },
 );
 
 sylt_macro::extern_function!(
     "sylt::lingon_sylt"
     l_bind_button
-    [Value::Int(controller), Value::String(button), Value::String(name)] -> Type::Void => {
+    "Binds a controller button to an input name"
+    [One(Int(controller)), One(String(button)), One(String(name))] -> Type::Void => {
         use lingon::input::{Device, Button};
         let button = if let Some(button) = Button::from_string(button) {
             button
         } else {
             return error!("l_bind_button", "'{}' is an invalid button", button);
         };
 
-        game!().input.bind(Device::Button(*controller as u32, button), String::clone(name));
-        Ok(Value::Nil)
+        game!().input.bind(Device::Button(*controller as u32, button), std::string::String::clone(name));
```

Comment:
Consider `name.clone()` instead, both here and above/below.

Reply:
I couldn't. And I got frustrated....